### PR TITLE
Auto-connect saved networks on start

### PR DESCRIPTION
**Pull request by Goshin**
- Connect saved networks on start.

Management for saved networks is not yet implemented, but it can be done in the `Keychain Access` App.

### CODE REVIEW

File: HeliPort/CredentialsManager.swift [hunk 1]
```diff
@@ -32,6 +33,17 @@ final class CredentialsManager {
...
+
+    func getSavedNetworks() -> [NetworkInfo] {
+        var list: [NetworkInfo] = []
+        for ssid in keychain.allKeys() {
+            // TODO: unserialize networkInfo
+            let network = NetworkInfo(ssid: ssid, connected: false, rssi: 0)
+            network.auth.security = NetworkInfo.AuthSecurity.CCMP.rawValue
+            list.append(network)
+        }
+        return list
```
Comment (igorkulman) [inline HeliPort/CredentialsManager.swift:45]:
```suggestion
        return keychain.allKeys().map { ssid in
            // TODO: unserialize networkInfo
            let network = NetworkInfo(ssid: ssid, connected: false, rssi: 0)
            network.auth.security = NetworkInfo.AuthSecurity.CCMP.rawValue
            return network
        }
```

bit more functional to keep the "theme" in the network manager class

File: HeliPort/CredentialsManager.swift [hunk 2]
```diff
@@ -18,19 +18,39 @@ final class CredentialsManager {
...
             return nil
         }
 
         Log.debug("Loading password for network \(network.ssid)")
-        return password
+        return try? JSONDecoder().decode(NetworkAuth.self, from: jsonData)
+    }
+
+    func getSavedNetworks() -> [NetworkInfo?] {
+        return keychain.allKeys().map { ssid in
```
Comment (igorkulman) [inline HeliPort/CredentialsManager.swift:46]: If you change `map` to `compactMap` the result will be a proper `[NetworkInfo]`, getting rid of the optionality

File: HeliPort/NetworkManager+Data.swift [hunk 3]
```diff
@@ -46,3 +46,19 @@ final class NetworkAuth {
...
+
+}
+
+extension NetworkAuth: Encodable {
+
+}
+
+extension NetworkAuth: Decodable {
+
+}
```
Comment (williambj1) [inline HeliPort/NetworkManager+Data.swift:64]: could be easily replaced with

```swift
final class NetworkAuth: Codable {
```

File: HeliPort/NetworkManager+Data.swift [hunk 4]
```diff
@@ -46,3 +46,19 @@ final class NetworkAuth {
...
 }
+
+extension NetworkInfo: Encodable {
+
+}
+
+extension NetworkInfo: Decodable {
+
+}
+
```
Comment (williambj1) [inline HeliPort/NetworkManager+Data.swift:57]: could be easily replaced with

```swift
final class NetworkInfo: Codable {
```

File: HeliPort/Supporting files/itl80211_security+Description.swift [hunk 5]
```diff
@@ -44,3 +44,11 @@ extension itl80211_security: CustomStringConvertible {
         }
     }
 }
+
+extension itl80211_security: Encodable {
```
Comment (igorkulman) [inline HeliPort/Supporting files/itl80211_security+Description.swift:48]: `Codable` is `Encodable` + `Decodable` so just conforming to this is enough

```suggestion
extension itl80211_security: Codable { }
```

Comment (Goshin) [reply HeliPort/Supporting files/itl80211_security+Description.swift:48]: Thanks, but seems the patch doesn't remove the following lines?

Comment (igorkulman) [reply HeliPort/Supporting files/itl80211_security+Description.swift:48]: You have to do a manual commit I guess. Hard to do in this Github web ui

**Pull request merged**